src/test/scala/org/broadinstitute/dsde/firecloud/service/LibraryApiServiceSpec.scala

```diff
@@ -368,5 +371,23 @@ class LibraryApiServiceSpec extends BaseServiceSpec with LibraryApiService with
         }
       }
     }
+
+    "when querying research purpose" - {
+      "POST with empty research purpose should return OK" in {
+        val request = ResearchPurposeRequest.empty
+        new RequestBuilder(HttpMethods.POST)(duosResearchPurposeQuery, request) ~> sealRoute(libraryRoutes) ~> check {
+          status should equal(OK)
+        }
+      }
+
+      "POST with disease focus should return OK" in {
+        val request = ResearchPurposeRequest.empty.copy(DS = Some(Seq(1234, 5678)))
+        new RequestBuilder(HttpMethods.POST)(duosResearchPurposeQuery, request) ~> sealRoute(libraryRoutes) ~> check {
+          status should equal(OK)
+          val diseaseIds = responseAs[JsObject].extract[Int]('bool / 'should / * / 'term / "structuredUseRestriction.DS" / 'value)
```

Comment:
It's almost like using something like XPath but for reaching into JSON.

Reply:
This is really cool - I haven not seen this before.